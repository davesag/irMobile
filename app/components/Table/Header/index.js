import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { Text } from 'react-native-elements'

import styles from '../styles'

const Header = ({ data }) =>
  data ? (
    <View style={styles.row}>
      {data.map(text => (
        <View key={text} style={styles.headerCell}>
          <Text style={styles.header}>{text}</Text>
        </View>
      ))}
    </View>
  ) : null

Header.propTypes = {
  data: PropTypes.arrayOf(PropTypes.string)
}

Header.defaultProps = {
  data: null
}

export default Header
