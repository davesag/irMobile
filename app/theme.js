import { Platform } from 'react-native'
import { colors } from 'react-native-elements'

export default {
  colors: {
    ...Platform.select({
      android: colors.platform.android,
      default: colors.platform.ios
    })
  }
}
