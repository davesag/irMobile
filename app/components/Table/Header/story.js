import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { SafeAreaView, View } from 'react-native'

import Header from '.'
import styles from '../styles'

const data = ['Aye', 'Bee', 'Sea', 'Dee']

storiesOf('Table.Header', module)
  .add('without header data', () => (
    <SafeAreaView>
      <View style={styles.table}>
        <Header />
      </View>
    </SafeAreaView>
  ))
  .add('with header data', () => (
    <SafeAreaView>
      <View style={styles.table}>
        <Header data={data} />
      </View>
    </SafeAreaView>
  ))
