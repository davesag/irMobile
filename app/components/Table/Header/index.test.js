import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import theme from '../../../theme'

import Header from '.'

const makeKey = (value, i) => i

let tree

describe('rendering', () => {
  describe('without data', () => {
    beforeAll(() => {
      tree = renderer.create(<Header theme={theme} makeKey={makeKey} />)
    })

    it('rendered correctly ', () => {
      expect(tree).toMatchSnapshot()
    })
  })

  describe('with data', () => {
    const data = ['Aye', 'Bee', 'Sea']

    beforeAll(() => {
      tree = renderer.create(
        <Header data={data} theme={theme} makeKey={makeKey} />
      )
    })

    it('rendered correctly ', () => {
      expect(tree).toMatchSnapshot()
    })
  })
})
