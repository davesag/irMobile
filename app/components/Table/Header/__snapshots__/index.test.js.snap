// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with data rendered correctly  1`] = `
<View>
  <View />
  <View />
  <View />
</View>
`;

exports[`rendering without data rendered correctly  1`] = `null`;
